README.md

```diff
@@ -1 +1,66 @@
 # AbstractDifferentiation
+
+## Motivation
+
+This is a package that implements an abstract interface for differentiation in Julia. This is particularly useful for implementing abstract algorithms requiring derivatives, gradients, jacobians, Hessians or multiple of those without depending on specific automatic differentiation packages' user interfaces.
+
+Julia has more (automatic) differentiation packages than you can count on 2 hands. Different packages have different user interfaces. Therefore, having a backend-agnostic interface to request the function value and its gradient for example is necessary to avoid a combinatorial explosion of code when trying to support every differentiation package in Julia in every algorithm package requiring gradients. For higher order derivatives, the situation is even more dire since you can combine any 2 differentiation backends together to create a new higher-order backend.
+
+## Loading `AbstractDifferentiation`
+
+To load `AbstractDifferentiation`, use:
+```julia
+using AbstractDifferentiation
+```
+`AbstractDifferentiation` exports a single name `AD` which is just an alias for the `AbstractDifferentiation` module itself. You can use this to access names inside `AbstractDifferentiation` using `AD.<>` instead of typing the long name `AbstractDifferentiation`.
```

Comment:
What's the point of this? Can't someone just do
```julia
import AbstractDifferentiation as AD
```
or 
```Julia
using AbstractDifferentiation
const AD = AbstractDifferentiation
```

Reply:
Yes. I am happy to remove this export if folks here agree that it's not useful. Technically, one can use `const` to alias any name in a package but I am not sure if this justifies not exporting anything at all. Exporting makes life easier, that's all. But I don't have strong opinions on this.